.github/workflows/continuous-integration-workflow.yml

```diff
@@ -76,36 +69,65 @@ jobs:
     steps:
       - name: Set JAVA_HOME to use Java 11
         run: echo "JAVA_HOME=$(echo $JAVA_HOME_11_X64)" >> $GITHUB_ENV
+      - name: Checkout local actions (that are invoked in the 'startCIServer' and 'stopCIServer' steps) # Do this _before_ untarring the distribution, as the checkout will empty the directory prior to the checkout!
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github
       - name: Download distribution artifact from build job.
         uses: actions/download-artifact@v4
         with:
           name: distribution-java11
-          path: distribution/target/distribution-base
-      - name: Download test files from build job.
-        uses: actions/download-artifact@v4
+          path: .
+      - name: untar distribution # sharing artifacts that consist of many files can be slow. Share one file instead.
+        run: tar -xf distribution-artifact.tar
+      - name: Checkout aioxmpp devel/head
+        uses: actions/checkout@v4
         with:
-          name: test-files-java11
-      - name: Fix file permissions
-        run: |
-          chmod +x distribution/target/distribution-base/bin/openfire.sh
-          chmod +x ./runAioxmppIntegrationTests
-
+          repository: horazont/aioxmpp
+          ref: devel
+          path: aioxmpp
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: 3.11
           check-latest: true # attempt to prevent to use 3.11.3 by enticing the runner to update (to something later)
-
+      - name: Install aoixmpp dependencies
+        run: python -m pip install setuptools pytest pytest-cov coveralls pyOpenSSL pytz
+      - name: Build aioxmpp
+        working-directory: ./aioxmpp
+        run: python -m pip install .
+      - name: Create Openfire config file for aioxmpp
+        working-directory: ./aioxmpp
+        run: |
+          cat >"openfire-config.ini" <<EOL
+          [global]
+          provisioner=aioxmpp.e2etest.provision.AnonymousProvisioner
+          
+          [aioxmpp.e2etest.provision.AnonymousProvisioner]
+          domain=example.org
+          host=localhost
+          port=5222
+          no_verify=true
+          quirks=["https://zombofant.net/xmlns/aioxmpp/e2etest/quirks#no-adhoc-ping", "https://zombofant.net/xmlns/aioxmpp/e2etest/quirks#no-xep-0049", "https://zombofant.net/xmlns/aioxmpp/e2etest/quirks#muc-no-333"]
+          EOL
```

Comment:
Should this be in the `.github` or `build/ci` directory and copied in, in case we want to run locally?

Reply:
Meh. Let's worry about that when we actually want to re-use it?